### ISSUE TITLE
Fails to parse method assignments with more than one arg and a return type.

### ISSUE THREAD

**Issue by sahrens** [reactjs/react-docgen#161]
this is fine:

```
class MyThing extends React.Component {
  foo = (a: number, b: number) => { // no return type annotation.
    return 5;
  };
  render() {
    return <div />;
  }
}
```

but this fails:

```
class MyThing extends React.Component {
  foo = (a: number, b: number): number => { // SyntaxError: Unexpected token
    return a * b;
  };
  render() {
    return <div />;
  }
}
```

These also work fine:

    foo = (a: number): number => {...};
    foo(a: number, b: number): number {...}

**Comment by fkling** [comment]
I assume this is also an issue with the parser version we are using. Hopefully this will be fixed once we can merge #158 .

**Comment by danez** [comment]
This should be fixed in 3.0
